logging/logrus/logger.go

```diff
@@ -224,37 +229,38 @@ func (logger *Logger) GetName() string {
 
 // SetOutput sets the standard logger output.
 func (logger *Logger) SetOutput(out io.Writer) {
-	logger.access.Lock()
-	defer logger.access.Unlock()
-	logger.std.Out = out
+	std := logger.stdStore.Load().(*lg.Logger)
+	std.Out = out
+	logger.stdStore.Store(std)
+
 }
 
 // SetFormatter sets the standard logger formatter.
 func (logger *Logger) SetFormatter(formatter lg.Formatter) {
-	logger.access.Lock()
-	defer logger.access.Unlock()
-	logger.std.Formatter = formatter
+	std := logger.stdStore.Load().(*lg.Logger)
+	std.Formatter = formatter
+	logger.stdStore.Store(std)
 }
 
 // SetLevel sets the standard logger level.
 func (logger *Logger) SetLevel(level logging.LogLevel) {
-	logger.access.Lock()
-	defer logger.access.Unlock()
+	std := logger.stdStore.Load().(*lg.Logger)
```

Comment:
I am not sure if this really helps since we are still updating level inside same *lg.Logger potentially by multiple go routines